### ISSUE TITLE
Two radio buttons sometimes can be selected simultaneously

### ISSUE THREAD

**Issue by sarchang** [phetsims/geometric-optics#68]
To reproduce: toggle back and forth between Marginal and Principal ray mode radio buttons, about 10 times. Occasionally both will be selected (circles turn blue). But only one ray mode is shown. 
Device: Macbook Air M1
Chrome Version 91.0.4472.114 (Official Build) (arm64)

```
assert.js:25 Uncaught Error: Assertion failed: alt key inconsistency between event and keyState.
    at window.assertions.assertFunction (assert.js:25)
    at KeyStateTracker.keydownUpdateAction.Action.phetioPlayback (KeyStateTracker.js:70)
    at Action.execute (Action.js:227)
    at GlobalKeyStateTracker.keydownUpdate (KeyStateTracker.js:148)
    at documentKeydownListener (KeyStateTracker.js:371)
window.assertions.assertFunction @ assert.js:25
KeyStateTracker.keydownUpdateAction.Action.phetioPlayback @ KeyStateTracker.js:70
execute @ Action.js:227
keydownUpdate @ KeyStateTracker.js:148
documentKeydownListener @ KeyStateTracker.js:371
assert.js:21 Assertion failed: alt key inconsistency between event and keyState.
assert.js:25 Uncaught Error: Assertion failed: alt key inconsistency between event and keyState.
    at window.assertions.assertFunction (assert.js:25)
    at KeyStateTracker.keydownUpdateAction.Action.phetioPlayback (KeyStateTracker.js:70)
    at Action.execute (Action.js:227)
    at GlobalKeyStateTracker.keydownUpdate (KeyStateTracker.js:148)
    at documentKeydownListener (KeyStateTracker.js:371)
window.assertions.assertFunction @ assert.js:25
KeyStateTracker.keydownUpdateAction.Action.phetioPlayback @ KeyStateTracker.js:70
execute @ Action.js:227
keydownUpdate @ KeyStateTracker.js:148
documentKeydownListener @ KeyStateTracker.js:371
assert.js:21 Assertion failed: alt key inconsistency between event and keyState.
assert.js:25 Uncaught Error: Assertion failed: alt key inconsistency between event and keyState.
    at window.assertions.assertFunction (assert.js:25)
    at KeyStateTracker.keydownUpdateAction.Action.phetioPlayback (KeyStateTracker.js:70)
    at Action.execute (Action.js:227)
    at GlobalKeyStateTracker.keydownUpdate (KeyStateTracker.js:148)
    at documentKeydownListener (KeyStateTracker.js:371)
window.assertions.assertFunction @ assert.js:25
KeyStateTracker.keydownUpdateAction.Action.phetioPlayback @ KeyStateTracker.js:70
execute @ Action.js:227
keydownUpdate @ KeyStateTracker.js:148
documentKeydownListener @ KeyStateTracker.js:371
```

**Comment by sarchang** [comment]
Was unable to reproduce in other simulations with radio buttons (CCK, Acid Base, Capacitor Lab).
Happened more than once in Geometric Optics

**Comment by sarchang** [comment]
@KatieWoe Can you reproduce this bug on Geometric Optics or on other devices/sims? We are using the standard vertical radio button group.

**Comment by KatieWoe** [comment]
I'm having trouble reproducing this in master on a Win 10 + Chrome device. Is there a particular build showing this issue?

**Comment by sarchang** [comment]
@KatieWoe I think it might have to do with me using the M1 Macbook Air + Chrome?

**Comment by KatieWoe** [comment]
We don't have an M1 Test device with QA, so I'm not able to confirm for sure. I also wasn't able to replicate it on our MacOS 11 device. I did notice that on master rays don't seem to be showing up at all, so I wonder if something changed on master.

**Comment by sarchang** [comment]
Rays should not appear unless you toggle the show/hide button. The rays seem to be working fine for me. And this issue seems to be difficult to reproduce; not sure where to go from here

**Comment by stemilymill** [comment]
@sarchang I have also been able to reproduce it, but like Katie I do not have a comparable device. Sorry I can't be more helpful!

**Comment by samreid** [comment]
I am unable to reproduce this problem on my M1 Macbook Air.  Are you testing with the mouse or keyboard?  The stack traces say a lot of things about key states and the alt key.

**Comment by kathy-phet** [comment]
@sarchang - can you capture with a screen video?

**Comment by jessegreenberg** [comment]
Indeed, the error is from KeyStateTracker, related to the "Alt" key. The error shouldn't happen, but was the "Alt" key pressed at any time this happened?

I wasn't able to see this happen on my Windows 10 machine, I switched between radio buttons frequently and pressed alt/ctrl/shift modifier keys frequently while doing so.

**Comment by jessegreenberg** [comment]
I just hit this error on my Windows 10 machine, but I am not sure how I got it and haven't been able to see it again. But since I was able to see it, it may not be platform specific.

**Comment by sarchang** [comment]
> @sarchang - can you capture with a screen video?

I am unable to reproduce it on my M1 Macbook Air as of now. I was using my trackpad @samreid. I was not pressing any of the  alt/crtl/shift modifier keys when clicking between the radio buttons.

**Comment by jessegreenberg** [comment]
@sarchang would you mind commenting which keyboard layout you use? Do you use a custom layout or one other than English (United States)?

**Comment by sarchang** [comment]
Nope it's just English (US).

**Comment by jessegreenberg** [comment]
OK thanks - The only reason I asked is because I could only get it to happen when I changed my keyboard layout. In those cases KeyStateTracker state is correct but the assertion is not, and I opened https://github.com/phetsims/scenery/issues/1249 to either remove this assertion or investigate a different way to implement it.

**Comment by Nancy-Salpepi** [comment]
I have an M1 MacBook Air and this happened to me today as well while using the most recent version of Chrome. I will try to reproduce.

**Comment by jessegreenberg** [comment]
I just pushed a fix for https://github.com/phetsims/scenery/issues/1249 and I can no longer get this to happen in my "United Kingdom" keyboard layout. I believe this is fixed, @Nancy-Salpepi can you please try again after this comment? If it cannot be observed now I think this can be closed.

**Comment by KatieWoe** [comment]
@Nancy-Salpepi make sure you test on master

**Comment by Nancy-Salpepi** [comment]
I'll work on that now and let you know.

**Comment by Nancy-Salpepi** [comment]
I wasn't able to replicate it.

**Comment by jessegreenberg** [comment]
OK great, thank you!

**Comment by sarchang** [comment]
Reopening because it happened to me again today. I was able to select two radio buttons simultaneously. It went back to normal as soon as I moved my cursor. I was not touching the keyboard. It happened again when assertions were enabled, but nothing was triggered. I suspect this has something to do with the cursor/display because at one point one and a half of the radio buttons were light blue and it looked like the color in the half-filled radio button was slowly spilling into the fully-filled radio button. I have also seen strange behavior in other areas of the sim, where moving the cursor changes what is displayed in the area immediately under the cursor. Many of these strange bugs are difficult to reproduce so we have been unable to figure out much. Unfortunately, I don't have any screenshots because they got lost in my clipboard history, but @veillette was on Zoom with me at the time and can confirm this behavior.

**Comment by samreid** [comment]
> where moving the cursor changes what is displayed in the area immediately under the cursor. 

This sounds like a dirty paint region issue, perhaps @jonathanolson knows more?

**Comment by Nancy-Salpepi** [comment]
Let me know if you want me to try again to replicate.

**Comment by jessegreenberg** [comment]
Thanks @Nancy-Salpepi and @sarchang. 

> like the color in the half-filled radio button was slowly spilling into the fully-filled radio button

Very odd. Also wondering if you have seen this outside of of Zoom, I have seen some cases where Zoom/screen sharing causes odd behaviors.

The assertion thrown in https://github.com/phetsims/geometric-optics/issues/68#issue-935977988 does not seem related to this anymore so removing my assignment, but feel free to pass it back if I can help.
